yaml.go

```diff
@@ -317,3 +317,63 @@ func convertToJSONableObject(yamlObj interface{}, jsonTarget *reflect.Value) (in
 		return yamlObj, nil
 	}
 }
+
+// JSONObjectToYAMLObject converts an in-memory JSON object into a YAML in-memory MapSlice,
+// without going through a byte representation.
+//
+// interface{} slices stay interface{} slices. map[string]interface{} becomes yaml.MapSlice.
+//
+// int64 and float64 are down casted following the logic of github.com/go-yaml/yaml:
+// - float64s are down-casted as far as possible without data-loss to int, int64, uint64.
+// - int64s are down-casted to int if possible without data-loss.
+//
+// Big int/int64/uint64 do not lose precision as in the json-yaml roundtripping case.
+//
+// string, bool and any other types are unchanged.
+func JSONObjectToYAMLObject(j map[string]interface{}) yaml.MapSlice {
+	if len(j) == 0 {
+		return nil
```

Comment:
this is surprising... doesn't this mean that an empty map doesn't round trip? if so, doesn't that break uses like CRD status (`status: {}`)

Reply:
It is what the yaml library does, compare the tests.

Reply:
hmm... it's still not quite clear to me who is intending to call this helper method, and whether this `len(0) => nil` treatment is going to bite us... any chance we can fix the yaml library instead?

Reply:
Not that this is just the top-level conversion func. Further down we have `jsonToYAMLValue` which is faithful with empty maps.

Reply:
@liggitt actually this semantics is not that surprising and nothing to change:

A yaml in-memory data structure has `interface{}{nil}` for null values. `yaml.MapSlice(nil)` is **not** null, but the empty map. It marshals to `{}`.

Reply:
Ah, ok